src/ba-transport.c

```diff
@@ -1733,11 +1736,69 @@ bool ba_transport_pcm_is_active(struct ba_transport_pcm *pcm) {
 
 int ba_transport_pcm_get_delay(const struct ba_transport_pcm *pcm) {
 	const struct ba_transport *t = pcm->t;
+	int delay = pcm->delay;
+
+	for (unsigned n = 0; n < pcm->delay_adjustments_count; n++) {
+		if (pcm->delay_adjustments[n].codec_id == t->codec_id) {
+			delay += pcm->delay_adjustments[n].adjustment;
+			break;
+		}
+	}
+
+	if (t->profile & BA_TRANSPORT_PROFILE_MASK_A2DP)
+		delay += t->a2dp.delay;
+	if (t->profile & BA_TRANSPORT_PROFILE_MASK_SCO)
+		delay += 10;
+
+	return delay;
+}
+
+void ba_transport_pcm_set_delay_adjustment(
+		struct ba_transport_pcm *pcm,
+		const char *codec_name,
+		int16_t adjustment) {
+
+	struct ba_transport *t = pcm->t;
+	uint16_t codec_id = 0xFFFF;
+
 	if (t->profile & BA_TRANSPORT_PROFILE_MASK_A2DP)
-		return t->a2dp.delay + pcm->delay;
+		codec_id = a2dp_codecs_codec_id_from_string(codec_name);
 	if (t->profile & BA_TRANSPORT_PROFILE_MASK_SCO)
-		return pcm->delay + 10;
-	return pcm->delay;
+		codec_id = hfp_codec_id_from_string(codec_name);
+	if (codec_id == 0xFFFF)
+		return;
+
+	for (size_t n = 0; n < pcm->delay_adjustments_count; n++) {
+		if (pcm->delay_adjustments[n].codec_id == codec_id) {
+			pcm->delay_adjustments[n].adjustment = adjustment;
+			return;
+		}
+	}
+
+	struct ba_transport_pcm_delay_adjustment *tmp = reallocarray(
```

Comment:
Maybe using `GArray` for the `delay_adjustment_count` will be easier? Or maybe event `GHashTable`....

Reply:
I have no previous experience with glib, but will have a play with both GArray and GHashTable to see what I can come up with.

Reply:
I've now changed the implementation to use GHashTable.